analysis/online/WhiterabbitCorrelationOnline.cxx

```diff
@@ -66,38 +73,140 @@ InitStatus WhiterabbitCorrelationOnline::Init()
+    h1_whiterabbit_correlation_bplast_ge->GetXaxis()->SetTitle("Time difference(bPlast - Germanium) [ns]");
+    h1_whiterabbit_correlation_bplast_ge->GetYaxis()->SetTitle("Counts");
+    h1_whiterabbit_correlation_bplast_ge->Draw();
+    folder_whiterabbit->Add(h1_whiterabbit_correlation_bplast_ge);
+    c_whiterabbit_correlation_bplast_ge->cd(0);
+
+    run->GetHttpServer()->RegisterCommand("Reset_Whiterabbit_Hist", Form("/Objects/%s/->Reset_Histo()", GetName()));
+    run->GetHttpServer()->RegisterCommand("Snapshot_Whiterabbit_Hist", Form("/Objects/%s/->Snapshot_Histo()", GetName()));
 
     return kSUCCESS;
     
 }
 
+// work in progress
 void WhiterabbitCorrelationOnline::Reset_Histo()
 {
-    c4LOG(info, "Resetting histograms!");
+    c4LOG(info, "I'm late, I'm late, for a very important date! No time to say 'Hello, Good Bye' I'm late, I'm late, I'm late!");
+    h1_whiterabbit_correlation_bplast_fatima->Reset();
+    h1_whiterabbit_correlation_fatima_ge->Reset();
+    h1_whiterabbit_correlation_bplast_ge->Reset();
+}
 
+void WhiterabbitCorrelationOnline::Snapshot_Histo()
+{
+    c4LOG(info, "Good heavens, would you look at the time!");
 
-}
+    //date and time stamp
+    time_t now = time(0);
+    tm *ltm = localtime(&now);
 
+    const char* snapshot_dir = Form("Whiterabbit_correlation_%d_%d_%d_%d_%d_%d.png", 1900 + ltm->tm_year, 1 + ltm->tm_mon, ltm->tm_mday, ltm->tm_hour, ltm->tm_min, ltm->tm_sec);
+    gSystem->mkdir(snapshot_dir);
+    gSystem->cd(snapshot_dir);
+
+    c_whiterabbit_correlation_bplast_fatima->SaveAs("c_whiterabbit_correlation_bplast_fatima.png");
+    c_whiterabbit_correlation_fatima_ge->SaveAs("c_whiterabbit_correlation_fatima_ge.png");
+    c_whiterabbit_correlation_bplast_ge->SaveAs("c_whiterabbit_correlation_bplast_ge.png");
+
+    gSystem->cd("..");
+
+    // snapshot .root file with data and time
+
+    file_whiterabbit_snapshot = new TFile(Form("Whiterabbit_correlation_%d_%d_%d_%d_%d_%d.root", 1900 + ltm->tm_year, 1 + ltm->tm_mon, ltm->tm_mday, ltm->tm_hour, ltm->tm_min, ltm->tm_sec), "RECREATE");
+    file_whiterabbit_snapshot->cd();
+    h1_whiterabbit_correlation_bplast_fatima->Write();
+    h1_whiterabbit_correlation_fatima_ge->Write();
+    h1_whiterabbit_correlation_bplast_ge->Write();
+    file_whiterabbit_snapshot->Close();
+    delete file_whiterabbit_snapshot;
+
+    gSystem->cd("..");
+    c4LOG(info, "Snapshot saved in:" << snapshot_dir);    
+}
 
 
 
 void WhiterabbitCorrelationOnline::Exec(Option_t* option)
 {   
-    if (fHitFatimaTwinpeaks && fHitFatimaTwinpeaks->GetEntriesFast() > 0)
+    //JB: there is probably a better way to code this. JEL or Nic hilfe bitte!
+    if (fHitFatimaTwinpeaks && fHitFatimaTwinpeaks->GetEntriesFast() > 0 && fHitbPlastTwinpeaks && fHitbPlastTwinpeaks->GetEntriesFast() > 0 && fHitGe && fHitGe->GetEntriesFast() > 0)
```

Comment:
At the moment this means the hits don't get filled unless Fatima+bPlast+HPGe come in coincidence together, which is pretty unlikely

One way may be to remove this if and have e.g.
```cpp
Int_t nHitsFatima = fHitFatimaTwinpeaks ? fHitFatimaTwinPeaks->GetEntriesFast() : 0;
```

Then the for loops just don't interate if the subsystem isn't there

IMO we should have a variable for the WR time that isn't in each hit, because for each event all hits have the same WR time (except AIDA) anyway, but this requires changing the xxReader.cxx

Reply:
I made the checks separate for each combination. I think this is more sensible.